lib/realtime/tenants/authorization.ex

```diff
@@ -130,40 +118,72 @@ defmodule Realtime.Tenants.Authorization do
 
   @policies_mods [ChannelPolicies, BroadcastPolicies, PresencePolicies]
   defp get_policies_for_connection(conn, authorization_context) do
-    Enum.reduce_while(@policies_mods, %Policies{}, fn policies_mod, policies ->
-      res =
-        Helpers.transaction(conn, fn transaction_conn ->
-          set_conn_config(transaction_conn, authorization_context)
-
-          get_policy_for_connection_and_feature(
-            policies_mod,
-            transaction_conn,
-            policies,
-            authorization_context
-          )
-        end)
+    Helpers.transaction(conn, fn transaction_conn ->
+      {:ok, _} =
+        Messages.create_message(
+          %{
+            channel_name: authorization_context.channel_name,
+            event: "event",
+            feature: :broadcast
+          },
+          transaction_conn
+        )
+
+      {:ok, _} =
+        Messages.create_message(
+          %{
+            channel_name: authorization_context.channel_name,
+            event: "event",
+            feature: :presence
+          },
+          transaction_conn
+        )
+
+      set_conn_config(transaction_conn, authorization_context)
+
+      policies = %Policies{}
+
+      policies =
+        get_read_policy_for_connection_and_feature(
+          transaction_conn,
+          policies,
+          authorization_context
+        )
+
+      policies =
+        get_write_policy_for_connection_and_feature(
+          transaction_conn,
+          policies,
+          authorization_context
+        )
+
+      Postgrex.query!(transaction_conn, "ROLLBACK AND CHAIN", [])
+
+      policies
+    end)
+  end
+
+  defp get_read_policy_for_connection_and_feature(conn, policies, authorization_context) do
+    Enum.reduce_while(@policies_mods, policies, fn policy_mod, policies ->
+      res = policy_mod.check_read_policies(conn, policies, authorization_context)
 
       case res do
         {:error, error} -> {:halt, {:error, error}}
         %DBConnection.TransactionError{} = err -> {:halt, err}
-        policy -> {:cont, policy}
+        {:ok, policy} -> {:cont, policy}
       end
     end)
   end
 
-  defp get_policy_for_connection_and_feature(
-         policy_mod,
-         transaction_conn,
-         policies,
-         authorization_context
-       ) do
-    with {:ok, policies} <-
-           policy_mod.check_write_policies(transaction_conn, policies, authorization_context),
-         {:ok, policies} <-
-           policy_mod.check_read_policies(transaction_conn, policies, authorization_context) do
-      policies
-    else
-      {:error, error} -> Postgrex.rollback(transaction_conn, error)
-    end
+  defp get_write_policy_for_connection_and_feature(conn, policies, authorization_context) do
```

Comment:
just to check if we ever need to debug this function at runtime because it's private

Reply:
we can get the same output from `Realtime.Tenants.Authorization.get_authorizations/2` in fact I will need to think on a good tool to test this in admin studio 👀